### PR TITLE
DM-41111 M2 bump test mtcs methods

### PR DESCRIPTION
**Pull request by psferguson**

### CODE REVIEW

File: doc/news/DM-41111.feature.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+creating mtcs methods to run the M2 bump test
```
Comment (tribeiro) [inline doc/news/DM-41111.feature.rst:1]: Please, add punctuation and capitalize the first letter in "creating". However, I'd rephrase it slightly though. Maybe something like,

```
Update ``MTCS`` class to support running M2 bump tests.
```

File: python/lsst/ts/observatory/control/maintel/mtcs.py [hunk 2]
```diff
@@ -1401,6 +1401,68 @@ async def is_m1m3_in_engineering_mode(self) -> bool:
...
+        """M2 actuator bump test.
+
+        Parameters
+        ----------
+        actuator : `int`
+            Id of the actuator.
+        period : `float`,
+            There will be two bumps and each bump will wait for (2 * period)
+            seconds.
+        force : `float`, optional
```
Comment (tribeiro) [inline python/lsst/ts/observatory/control/maintel/mtcs.py:1419]: Is this really optional? If so, what is the default value?

File: python/lsst/ts/observatory/control/maintel/mtcs.py [hunk 3]
```diff
@@ -1401,6 +1401,68 @@ async def is_m1m3_in_engineering_mode(self) -> bool:
 
         return m1m3_detailed_state in self.m1m3_engineering_states
 
+    async def run_m2_actuator_bump_test(
+        self,
+        actuator: int,
+        period: float,
```
Comment (tribeiro) [inline python/lsst/ts/observatory/control/maintel/mtcs.py:1407]: Consider adding a default value for the period.

File: python/lsst/ts/observatory/control/maintel/mtcs.py [hunk 4]
```diff
@@ -1401,6 +1401,68 @@ async def is_m1m3_in_engineering_mode(self) -> bool:
...
+        force : `float`, optional
+            the +/- push/pull foce to be applied in N
+        """
+        # check that actuator_id is not hardpoint
+        hardpoint_ids = await self.get_m2_hardpoints()
+
+        # csc actuator id is 0 based and hardpoint id is 1 based
+        if actuator + 1 in hardpoint_ids:
+            raise Exception(
+                f"Cannot bump test one of the M2 hardpoints: actuator = {actuator}"
```
Comment (tribeiro) [inline python/lsst/ts/observatory/control/maintel/mtcs.py:1428]: Please, add a "." at the end of the phrase.

File: python/lsst/ts/observatory/control/maintel/mtcs.py [hunk 5]
```diff
@@ -1401,6 +1401,68 @@ async def is_m1m3_in_engineering_mode(self) -> bool:
...
+        # bump test single actuator
+        await self.rem.mtm2.cmd_actuatorBumpTest.set_start(
+            actuator=actuator,
+            period=period,
+            force=force,
+        )
+
+    async def get_m2_hardpoints(
+        self,
+    ) -> list:
```
Comment (tribeiro) [inline python/lsst/ts/observatory/control/maintel/mtcs.py:1440]: I think you can specify that this is a list of ints, something like `list[int]`, otherwise mypy assumes it is a list of anything.

File: python/lsst/ts/observatory/control/maintel/mtcs.py [hunk 6]
```diff
@@ -1401,6 +1401,68 @@ async def is_m1m3_in_engineering_mode(self) -> bool:
...
+        """Retrieve the current list of M2 hardpoints.
+
+        Returns
+        -------
+        list
```
Comment (tribeiro) [inline python/lsst/ts/observatory/control/maintel/mtcs.py:1445]:
```
`list` [ `int` ]
```

File: python/lsst/ts/observatory/control/mock/mtcs_async_mock.py [hunk 7]
```diff
@@ -153,6 +153,7 @@ async def setup_types(self) -> None:
 
         # MTM2 data
         self._mtm2_evt_force_balance_system_status = types.SimpleNamespace(status=False)
+        self._mtm2_evt_hardpointList = types.SimpleNamespace(actuators=[6])
```
Comment (tribeiro) [inline python/lsst/ts/observatory/control/mock/mtcs_async_mock.py:156]: I think you mean 

```
self._mtm2_evt_hardpointList = types.SimpleNamespace(actuators=list(range(6)))
```

**Review by tribeiro — COMMENTED** [review]
Looking good. Here some comments for you to consider before I can approve it.

File: python/lsst/ts/observatory/control/maintel/mtcs.py (hunk 2 above)
Comment (psferguson) [reply python/lsst/ts/observatory/control/maintel/mtcs.py:1419]: removed the optional

File: python/lsst/ts/observatory/control/maintel/mtcs.py (hunk 3 above)
Comment (psferguson) [reply python/lsst/ts/observatory/control/maintel/mtcs.py:1407]: Set a default time of 60 seconds

File: python/lsst/ts/observatory/control/mock/mtcs_async_mock.py (hunk 7 above)
Comment (psferguson) [reply python/lsst/ts/observatory/control/mock/mtcs_async_mock.py:156]: why the `range(6)`? 
I could copy the list they give here:
https://github.com/lsst-ts/ts_m2/blob/develop/tests/test_csc.py#L157-L160

Comment (tribeiro) [reply python/lsst/ts/observatory/control/mock/mtcs_async_mock.py:156]: right.. that is probably a better idea.. I naively thought it was from 0-5....

**Review by tribeiro — APPROVED** [review]
Looks good!
